Running individual tests is broken, `cargo build` and `cargo test` still work

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. `cargo test --package harper-core --lib -- linting::modal_of::tests::doesnt_catch_to_take_on_the_full_might_of_nato --exact --show-output`
  (or any other individual test, or from your IDE's runner)
3. See if output looks like this:
```
   Compiling harper-core v0.31.0 (/Users/hippietrail/harper-the-second/harper/harper-core)
error[E0412]: cannot find type `NonZeroUsize` in this scope
   --> harper-core/src/patterns/mod.rs:166:68
    |
166 |     fn matches(&self, tokens: &[Token], source: &[char]) -> Option<NonZeroUsize> {
    |                                                                    ^^^^^^^^^^^^ not found in this scope
    |
help: consider importing this type alias
    |
8   + use std::num::NonZeroUsize;
    |

For more information about this error, try `rustc --explain E0412`.
error: could not compile `harper-core` (lib test) due to 1 previous error
```

**Screenshots**

<img width="1050" alt="Image" src="https://github.com/user-attachments/assets/b4fc3e60-b9fc-430b-9a27-3678f3a28901" />

**Platform**
Windsurf on macOS

**Additional context**

We had a similar issue a few days ago that was fixed. It's due to how tests can be run concurrently or not via some settings and conditional code I believe. @RunDevelopment 

----

@elijah-potter Do you think we should add a `build` GitHub label?

--- a/harper-core/src/patterns/mod.rs
+++ b/harper-core/src/patterns/mod.rs
@@ -163,7 +163,7 @@ impl<F> Pattern for F
 where
     F: Fn(&Token, &[char]) -> bool,
 {
-    fn matches(&self, tokens: &[Token], source: &[char]) -> Option<NonZeroUsize> {
+    fn matches(&self, tokens: &[Token], source: &[char]) -> Option<usize> {
         if self(tokens.first()?, source) {
             Some(1)
         } else {

